scripts/ci-report/fetch_stats_from_ci.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Usage: ./fetch_stats_from_ci.sh
+#
+# This script is for fetching historic test stats from GitHub Actions CI.
+#
+# Requires gh with credentials.
+#
+# https://github.com/cli/cli/blob/trunk/pkg/cmd/run/view/view.go#L434
+
+dir="$(dirname "$0")"/ci-stats
+mkdir -p "${dir}"
+
+pushd "${dir}" >/dev/null
+
+# Stats step name, used for filtering log.
+job_step_name="Print test stats"
+
+if [[ ! -f list-ci.yaml.json ]]; then
+	gh run list -w ci.yaml -L 1000 --json conclusion,createdAt,databaseId,displayTitle,event,headBranch,headSha,name,number,startedAt,status,updatedAt,url,workflowDatabaseId,workflowName \
+		>list-ci.yaml.json || {
+		rm -f list-ci.yaml.json
+		exit 1
+	}
+fi
+
+runs="$(
+	jq -r '.[] | select(.status == "completed") | select(.conclusion == "success" or .conclusion == "failure") | [.databaseId, .event, .displayTitle, .headBranch, .headSha] | @tsv' \
+		<list-ci.yaml.json
+)"
+
+while read -r run; do
+	mapfile -d $'\t' -t parts <<<"${run}"
+	parts[-1]="${parts[-1]%$'\n'}"
+
+	database_id="${parts[0]}"
+	event="${parts[1]}"
+	display_title="${parts[2]}"
+	head_branch="${parts[3]}"
+	head_sha="${parts[4]}"
+
+	if [[ ${database_id} -le 4595490577 ]]; then
```

Comment:
nit: Maybe add a comment explaining what is this magic number

Reply:
Hmm, I guess the print below is a bit vague, "too old", I'll add a comment.